pkg/api/v1alpha1/vspheredatacenterconfig_webhook.go

```diff
@@ -81,7 +82,7 @@ func (r *VSphereDatacenterConfig) ValidateUpdate(old runtime.Object) error {
 	allErrs = append(allErrs, validateImmutableFieldsVSphereCluster(r, oldDatacenterConfig)...)
 
 	if len(allErrs) == 0 {
-		return nil
+		return r.Validate()
```

Comment:
why don't we wrap the result of `Validate` in an apierror like we do in other webhooks?